src/functional/iterator/ir.py

```diff
@@ -24,20 +24,9 @@ class Expr(Node):
     ...
 
 
-class BoolLiteral(Expr):
-    value: bool
-
-
-class IntLiteral(Expr):
-    value: int
-
-
-class FloatLiteral(Expr):
-    value: float  # TODO other float types
-
-
-class StringLiteral(Expr):
+class Literal(Expr):
     value: str
+    type: str  # noqa: A003
```

Comment:
```suggestion
    type: Literal["bool", "int", "float"]  # noqa: A003
```

Reply:
This was discussed with @fthaler who did not see a reason to restrict this. Also just to support field view, "int32", "int64", "float32", "float64" would have to be in there.